hydra-node/src/Hydra/API/Server.hs

```diff
@@ -117,22 +125,57 @@ runAPIServer ::
   (IsChainState tx) =>
   IP ->
   PortNumber ->
+  Party ->
   Tracer IO APIServerLog ->
   TVar [TimedServerOutput tx] ->
   (ClientInput tx -> IO ()) ->
   TChan (TimedServerOutput tx) ->
   IO ()
-runAPIServer host port tracer history callback responseChannel = do
+runAPIServer host port party tracer history callback responseChannel = do
   traceWith tracer (APIServerStarted port)
   handle onIOException $
     runServer (show host) (fromIntegral port) $ \pending -> do
+      let path = requestPath $ pendingRequest pending
+      queryParams <- uriQuery <$> mkURIBs path
       con <- acceptRequest pending
       chan <- STM.atomically $ dupTChan responseChannel
       traceWith tracer NewAPIConnection
-      forwardHistory con
+
+      -- api client can decide if they want to see the past history of server outputs
+      dontServeHistory <- shouldNotServeHistory queryParams
+      if dontServeHistory
+        then forwardGreetingOnly con
+        else forwardHistory con
```

Comment:
I would use a positive form _shouldServeHistory_ instead of a negative one so that we say what we want and not what we don't want and we don't do to much of double negations by negating that we don't want what we don't want... or want? Not sure ;)
```suggestion
      serveHistory <- shouldServeHistory queryParams
      if serveHistory
        then forwardHistory con
        else forwardGreetingOnly con 
```

Reply:
:) I did it this way because serving history is a default and client can decide if they don't want to see history.